config/consensus.go

```diff
@@ -391,6 +391,9 @@ type ConsensusParams struct {
 	EnableKeyregCoherencyCheck bool
 
 	EnableExtraPagesOnAppUpdate bool
+
+	// Allow querying txn.FirstValid - 1 timestamp
+	EnableFirstValidTimeStamp bool
```

Comment:
Does this need to be separate from enabling the field in Teal 6?

Reply:
I'm a bit worried on erring on EvalParam preparation, and we have not accessed old blocks there before. It should work but seemed to be better on a safe side. I like the idea if returning zero and erring in runtime instead of prep time

Reply:
removed, not needed anymore since `BlockTimeStamp` does not err anymore